File: test-suite-caffeine-java/src/test/java/io/micronaut/cache/CounterService.java
package io.micronaut.cache;

import io.micronaut.cache.annotation.CacheConfig;
import io.micronaut.cache.annotation.CacheInvalidate;
import io.micronaut.cache.annotation.CachePut;
import io.micronaut.cache.annotation.Cacheable;
import jakarta.inject.Singleton;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * @author Graeme Rocher
 * @since 1.0
 */
@Singleton
@CacheConfig(cacheNames = {"counter"})
public class CounterService {

    Map<String, Integer> counters = new LinkedHashMap<>();

    @CachePut
    public int increment(String name) {
        int value = counters.computeIfAbsent(name, s -> 0);
        counters.put(name, ++value);
        return value;
    }

    @Cacheable
    public int getValue(String name) {
        return counters.computeIfAbsent(name, s -> 0);
    }


    @CacheInvalidate()
    public void reset(String name) {
        counters.remove(name);
    }

    @CacheInvalidate
    public void set(String name, int val) {
        counters.put(name, val);
    }
}

File: test-suite-hazelcast-python/src/test/python/micronaut/cache/docs/CounterService.py
from jakarta.inject import Singleton
from micronaut.cache.annotation import CacheConfig, CacheInvalidate, CachePut, Cacheable


@Singleton
@CacheConfig(cacheNames=["counter"])
class CounterService:

    def __init__(self):
        self.counters: dict[str, int] = {}

    @CachePut
    def increment(self, name: str) -> int:
        value = self.counters.get(name, 0) + 1
        self.counters[name] = value
        return value

    @Cacheable
    def get_value(self, name: str) -> int:
        return self.counters.setdefault(name, 0)

    @CacheInvalidate
    def reset(self, name: str) -> None:
        self.counters.pop(name, None)

    @CacheInvalidate
    def set(self, name: str, val: int) -> None:
        self.counters[name] = val
